pkg/network/http/driver_interface.go

```diff
@@ -0,0 +1,238 @@
+	if err != nil {
+		return nil, err
+	}
+
+	var filters []driver.FilterDefinition
+	for _, iface := range ifaces {
+		// IPv4
+		filters = append(filters, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionOutbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET,
+			Protocol:       windows.IPPROTO_TCP,
+		}, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionInbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET,
+			Protocol:       windows.IPPROTO_TCP,
+		})
+
+		// IPv6
+		filters = append(filters, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionOutbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET6,
+			Protocol:       windows.IPPROTO_TCP,
+		}, driver.FilterDefinition{
+			FilterVersion:  driver.Signature,
+			Size:           driver.FilterDefinitionSize,
+			Direction:      driver.DirectionInbound,
+			FilterLayer:    driver.LayerTransport,
+			InterfaceIndex: uint64(iface.Index),
+			Af:             windows.AF_INET6,
+			Protocol:       windows.IPPROTO_TCP,
+		})
+	}
+
+	return filters, nil
+}
+
+func (di *httpDriverInterface) setMaxFlows(maxFlows uint64) error {
+	log.Debugf("Setting max flows in driver http filter to %v", maxFlows)
+	err := windows.DeviceIoControl(di.driverHTTPHandle.Handle,
+		driver.SetMaxFlowsIOCTL,
+		(*byte)(unsafe.Pointer(&maxFlows)),
+		uint32(unsafe.Sizeof(maxFlows)),
+		nil,
+		uint32(0), nil, nil)
+	if err != nil {
+		log.Warnf("Failed to set max number of flows in driver http filter to %v %v", maxFlows, err)
+	}
+	return err
+}
+
+func (di *httpDriverInterface) startReadingBuffers() {
+	di.eventLoopWG.Add(1)
+	go func() {
+		defer di.eventLoopWG.Done()
+
+		transactionSize := uint32(C.size_of_http_transaction_type())
+		for {
+			buf, bytesRead, err := driver.GetReadBufferWhenReady(di.iocp)
+			if iocpIsClosedError(err) {
+				return
+			}
+			if err != nil {
+				log.Infof("Error reading http transaction buffer: %v", err)
+				continue
+			}
+
+			batchSize := bytesRead / transactionSize
+			transactionBatch := make([]driver.HttpTransactionType, batchSize)
+
+			for i := uint32(0); i < batchSize; i++ {
+				transaction := (*driver.HttpTransactionType)(unsafe.Pointer(&buf.Data[i*transactionSize]))
+				deepCopyTransactionData(&transactionBatch[i], transaction)
+			}
+
+			di.dataChannel <- transactionBatch
+
+			err = driver.StartNextRead(di.driverHTTPHandle.Handle, buf)
+			if err != nil && err != windows.ERROR_IO_PENDING {
+				log.Infof("Error starting next http transaction read: %v")
```

Comment:
Is `INFO` the appropriate log level? I think either going to the warn/error side, or the debug side makes sense here.